bases/renku_data_services/data_api/main.py

```diff
@@ -148,6 +148,13 @@ async def setup_rclone_validator(app: Sanic) -> None:
         validator = RCloneValidator()
         app.ext.dependency(validator)
 
+    @app.before_server_start
+    async def start_k8s_cache_synchronization(app: Sanic) -> None:
+        sync_coroutines = config.nb_config.k8s_client.get_synchronization_coroutines()
+        if config.builds_config.enabled and config.builds_config.shipwright_client is not None:
+            sync_coroutines.extend(config.builds_config.shipwright_client.client.get_synchronization_coroutines())
+        app.add_task(asyncio.gather(*sync_coroutines), name="k8s_cache_sync")
+
```

Comment:
question: why run this in main data services instead of k8s watcher or data_tasks jobs? This will run on each horizontally scaled data-svc deployment, not sure that's what we want.

I'd personally just add it in the loop here: https://github.com/SwissDataScienceCenter/renku-data-services/blob/main/components/renku_data_services/k8s_watcher/core.py#L103 (gated by an `if datetime.now() <last_sync + interval:` kinda thing) then there's no concurrency issue or anything like that. and code handling each kind can be kinda self-contained

Reply:
Yeah I will move it. I thought I had a reason but I cannot think of it now.

Reply:
done - you can look at the 2nd of the two new commits to see the changes